rskj-core/src/main/java/co/rsk/core/bc/ParallelizeTransactionHandler.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.core.bc;
+
+import co.rsk.core.RskAddress;
+import org.ethereum.core.Transaction;
+import org.ethereum.db.ByteArrayWrapper;
+import org.ethereum.vm.GasCost;
+
+import java.util.*;
+
+public class ParallelizeTransactionHandler {
+    private final TransactionBucket sequentialBucket;
+    private final HashMap<ByteArrayWrapper, Set<Short>> bucketByWrittenKey;
```

Comment:
can a key be written in multiple buckets?

Reply:
Two different transactions that write/read the same key can be in both Sequential and Parallel buckets.

Reply:
Nope, you were right